status/status.go

```diff
@@ -37,6 +39,147 @@ import (
 // NamePrefix is a common name prefix for this service.
 const NamePrefix = "Red Hat Trusted App Test"
 
+// ScenarioReportStatus keep report status of git provider for the particular scenario
+type ScenarioReportStatus struct {
+	LastUpdateTime *time.Time `json:"lastUpdateTime"`
+}
+
+// SnapshotReportStatus keep report status of git provider for the snapshot
+type SnapshotReportStatus struct {
+	Scenarios map[string]*ScenarioReportStatus `json:"scenarios"`
+	dirty     bool
+}
+
+// SetLastUpdateTime updates the last udpate time of the given scenario to the given time
+func (srs *SnapshotReportStatus) SetLastUpdateTime(scenarioName string, t time.Time) {
+	srs.dirty = true
+	if scenario, ok := srs.Scenarios[scenarioName]; ok {
+		scenario.LastUpdateTime = &t
+		return
+	}
+
+	srs.Scenarios[scenarioName] = &ScenarioReportStatus{
+		LastUpdateTime: &t,
+	}
+}
+
+// IsNewer returns true if given scenario has newer time than the last updated
+func (srs *SnapshotReportStatus) IsNewer(scenarioName string, t time.Time) bool {
+	if scenario, ok := srs.Scenarios[scenarioName]; ok {
+		return scenario.LastUpdateTime.Before(t)
+	}
+
+	// no record, it's new
+	return true
+}
+
+// ToAnnotationString exports data in format for annotation
+func (srs *SnapshotReportStatus) ToAnnotationString() (string, error) {
+	byteVar, err := json.Marshal(srs)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal report status metadata: %w", err)
+	}
+	return string(byteVar), nil
+}
+
+// IsDirty returns true if there are new changes to be written
+func (srs *SnapshotReportStatus) IsDirty() bool {
+	return srs.dirty
+}
+
+// ResetDirty marks changes as synced to snapshot
+func (srs *SnapshotReportStatus) ResetDirty() {
+	srs.dirty = false
+}
+
+// NewSnapshotReportStatus creates new object
+func NewSnapshotReportStatus(jsondata string) (*SnapshotReportStatus, error) {
+	srs := SnapshotReportStatus{
+		Scenarios: map[string]*ScenarioReportStatus{},
+	}
+	if jsondata != "" {
+		err := json.Unmarshal([]byte(jsondata), &srs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal json: %w", err)
+		}
+	}
+	return &srs, nil
+}
+
+// NewSnapshotReportStatusFromSnapshot creates new SnapshotTestStatus struct from snapshot annotation
+func NewSnapshotReportStatusFromSnapshot(s *applicationapiv1alpha1.Snapshot) (*SnapshotReportStatus, error) {
+	var (
+		statusAnnotation string
+		ok               bool
+	)
+
+	if statusAnnotation, ok = s.ObjectMeta.GetAnnotations()[gitops.SnapshotStatusReportAnnotation]; !ok {
+		statusAnnotation = ""
+	}
+
+	srs, err := NewSnapshotReportStatus(statusAnnotation)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get integration tests report status metadata from snapshot %s: %w", s.Name, err)
```

Comment:
suggestion: also adding the annotation name here so it's clearer when debugging the issue